src/units/abilities.cpp

```diff
@@ -1262,6 +1262,9 @@ std::pair<std::string, std::string> attack_type::damage_type() const
 std::set<std::string> attack_type::alternative_damage_types() const
 {
 	unit_ability_list damage_alternative_type_list = get_specials_and_abilities("damage_type");
+	utils::erase_if(damage_alternative_type_list, [&](const unit_ability& i) {
+		return !(*i.ability_cfg).has_attribute("alternative_type");
+	});
```

Comment:
Since we loop over the list below anyway it'd be better to add the `has_attribute` check there.

Reply:
corrected and change name of list